ribs/emitters/opt/_cma_es.py

```diff
@@ -216,14 +219,14 @@ def ask(self, lower_bounds, upper_bounds, batch_size=None):
             new_solutions, out_of_bounds = self._transform_and_check_sol(
                 unscaled_params, transform_mat, self.mean, lower_bounds,
                 upper_bounds)
-            solutions[remaining_indices] = new_solutions
+            self._solutions[remaining_indices] = new_solutions
 
             # Find indices in remaining_indices that are still out of bounds
             # (out_of_bounds indicates whether each value in each solution is
             # out of bounds).
             remaining_indices = remaining_indices[np.any(out_of_bounds, axis=1)]
 
-        return np.asarray(solutions)
+        return self._solutions
```

Comment:
I don't think it's a good idea to return a member variable here, but a copy also seems needlessly expensive. Let's return a readonly array -- see the readonly function here https://github.com/icaros-usc/pyribs/blob/master/ribs/archives/_archive_base.py#L20

Reply:
Also note this for the other emitters -- we can move `readonly` to `ribs._utils`